src/Controls/src/Core/SetterSpecificity.cs

```diff
@@ -1,102 +1,172 @@
+		/// </summary>
+		/// <param name="vsm">
+		/// Specifies if the value comes from VisualStateManager <br />
+		///	- 0: not from VSM <br />
+		/// - 1: from VSM <br />
+		/// - 0xFF: from Handler
+		/// </param>
+		/// <param name="manual">
+		/// Determines manual specificity, also covers triggers <br />
+		/// - 0: not manual <br />
+		/// - 1..100: manual <br />
+		/// - 101..N: triggers <br />
+		/// </param>
+		/// <param name="dynamicResource">Set to 1 when value comes from dynamic resource, otherwise 0</param>
+		/// <param name="binding">Set to 1 when value comes from binding, otherwise 0</param>
+		/// <param name="style">
+		/// XAML Style specificity <br />
+		/// - 0: not from Style <br />
+		/// - 100-n: implicit style, n = max(99, distance between the RD and the target) <br />
+		/// - 200: local style, inline css, <br />
+		/// - 300-n: RD Style, n = max(99, distance between the RD and the target) <br />
+		/// - 400-n: VSM, n = max(99, distance between the RD and the target) <br />
+		/// - 500: !important (not implemented) <br />
+		/// </param>
+		/// <param name="id">
+		/// CSS Id Specificity <br />
+		/// See https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
+		/// </param>
+		/// <param name="class">
+		/// CSS Class Specificity <br />
+		/// See https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
+		/// </param>
+		/// <param name="type">
+		/// CSS Type Specificity <br />
+		/// See https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
+		/// </param>
+		/// <exception cref="ArgumentOutOfRangeException"></exception>
+		public SetterSpecificity(byte vsm, ushort manual, byte dynamicResource, byte binding, ushort style, byte id, byte @class, byte type)
+		{
+			if (style > 0xFFE) throw new ArgumentOutOfRangeException(nameof(style));
+			if (dynamicResource > 1) throw new ArgumentOutOfRangeException(nameof(dynamicResource));
+			if (binding > 1) throw new ArgumentOutOfRangeException(nameof(binding));
+			
+			// If no style is set, set it to a value which supersedes any other style value
+			if (style == 0)
+			{
+				style = 0xFFF;
+				id = @class = type = 0xFF;
+			}
+
+			// Priority order:
+			//                       64bit ulong value
+			// 1. VSM :              0xFF00000000000000
+			// 2. Style              0x00FFF00000000000
+			// 3. Manual(& Trigger)  0x00000FFFF0000000
+			// 4. DynamicResource    0x0000000002000000
+			// 5. Binding            0x0000000001000000
+			// 6. Id                 0x0000000000FF0000
+			// 7. Class              0x000000000000FF00
+			// 8. Type               0x00000000000000FF
+			
+			// Combine dynamicResource and binding into a single byte
+			var dynamicResourceBinding = (dynamicResource * 0x2) | (binding * 0x1);
+			
+			_value = type
+			         | (ulong)@class << 8
+			         | (ulong)id << 16
+			         | (ulong)dynamicResourceBinding << 24
+			         | (ulong)manual << 28
+			         | (ulong)style << 44
+			         | (ulong)vsm << 56;
```

Comment:
We cannot do that, because `<<` operator implicitly casts `byte` to `int` which will not support as much shifts as we need.
We have many proofs of this, starting from compiler warning and errors (depending on the syntax being used).
As double-check, this test fails on the last line.
```cs
[Fact]
public void CastUlongWorksFine()
{
	byte x000FF00000000000 = 1;
	byte x0000000000FF0000 = 1;

	ulong expected = 0x0000100000010000;
	ulong actualWithMultiCasts = (ulong)x000FF00000000000 << 44 | (ulong)x0000000000FF0000 << 16;
	ulong actualWithSingleCast = (ulong)(x000FF00000000000 << 44 | x0000000000FF0000 << 16);
	Assert.Equal(expected, actualWithMultiCasts);
	Assert.Equal(expected, actualWithSingleCast);
}
```